internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/BlobP2PTransferAmongServersTest.java

```diff
@@ -0,0 +1,163 @@
+import java.nio.file.Paths;
+import java.util.Properties;
+import java.util.function.Consumer;
+import org.apache.commons.io.FileUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class BlobP2PTransferAmongServersTest {
+  private static final Logger LOGGER = LogManager.getLogger(BlobP2PTransferAmongServersTest.class);
+
+  private String path1;
+  private String path2;
+
+  @Test(singleThreaded = true)
+  public void testBlobP2PTransferAmongServers() throws Exception {
+    // set up store with 2 servers and trigger the job
+    VeniceClusterWrapper cluster = initializeVeniceCluster();
+    String storeName = "test-store";
+    Consumer<UpdateStoreQueryParams> paramsConsumer = params -> params.setBlobTransferEnabled(true);
+    setUpStore(cluster, storeName, paramsConsumer, properties -> {}, true);
+
+    VeniceServerWrapper server1 = cluster.getVeniceServers().get(0);
+    VeniceServerWrapper server2 = cluster.getVeniceServers().get(1);
+
+    // very the snapshot is generated for both servers after the job is done
+    for (int i = 0; i < 3; i++) {
+      String snapshotPath1 = RocksDBUtils.composeSnapshotDir(path1 + "/rocksdb", storeName + "_v1", i);
+      Assert.assertTrue(Files.exists(Paths.get(snapshotPath1)));
+      String snapshotPath2 = RocksDBUtils.composeSnapshotDir(path2 + "/rocksdb", storeName + "_v1", i);
+      Assert.assertTrue(Files.exists(Paths.get(snapshotPath2)));
+    }
+
+    // cleanup and restart server 1
+    FileUtils.deleteDirectory(
+        new File(RocksDBUtils.composePartitionDbDir(path1 + "/rocksdb", storeName + "_v1", METADATA_PARTITION_ID)));
+    FileUtils.deleteDirectory(new File(RocksDBUtils.composePartitionDbDir(path1 + "/rocksdb", storeName + "_v1", 0)));
+    FileUtils.deleteDirectory(new File(RocksDBUtils.composePartitionDbDir(path1 + "/rocksdb", storeName + "_v1", 1)));
+    FileUtils.deleteDirectory(new File(RocksDBUtils.composePartitionDbDir(path1 + "/rocksdb", storeName + "_v1", 2)));
+    cluster.stopAndRestartVeniceServer(server1.getPort());
+
+    // wait 2 min
+    Thread.sleep(120000);
+
+    // the partition files should be transferred to server 1 and offset should be the same
+    for (int i = 0; i < 3; i++) {
+      File file = new File(RocksDBUtils.composePartitionDbDir(path1 + "/rocksdb", storeName + "_v1", i));
+      Boolean fileExisted = Files.exists(file.toPath());
+      Assert.assertTrue(fileExisted);
+
+      OffsetRecord offsetServer1 =
+          server1.getVeniceServer().getStorageMetadataService().getLastOffset("test-store_v1", i);
+      OffsetRecord offsetServer2 =
+          server2.getVeniceServer().getStorageMetadataService().getLastOffset("test-store_v1", i);
+      Assert.assertEquals(offsetServer1.getLocalVersionTopicOffset(), offsetServer2.getLocalVersionTopicOffset());
+    }
+  }
+
+  public VeniceClusterWrapper initializeVeniceCluster() {
```

Comment:
The practice we usually do it to move codes related to setup env to a method, say `setup`, then add testNG annotation to let it run before test cases so we can potentially add more test cases for this suite. Can you plz do that?